src/f0/app_solar/source/tmp101.c

```diff
@@ -0,0 +1,261 @@
+
+bool read_tmp101an_temperature_v1(TMP101Driver *devp, unsigned int option)
+{
+// 2021-02-07 - these two byte arrays will be passed to a development
+//  routine, to confirm we can talk with the TMP101AN sensors of the
+//  solar board.  When that's working we'll move these and the dev'
+//  code over to the TMP101AN driver source file.  That will be the
+//  first step to honor and conform to the object oritented device
+//  driver encapsulation that's already expressed in the Oresat firmware
+//  project - TMH
+// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+    uint8_t buffer_tx[2] = {0};
+    uint8_t buffer_rx[5] = {0};
+
+    msg_t i2c_result = MSG_OK;
+    i2cflags_t i2c_flags = 0;
+
+    uint8_t sensor_addr = devp->config->saddr;
+
+    if (option == 2)
+    {
+        // option parameter not yet used - TMH
+    }
+
+// Values chosen per TMP101AN datasheet:
+    buffer_tx[0] = TMP101_REG_TEMPERATURE_READING;
+    buffer_tx[1] = 0;
+
+// Clear first two bytes of receive buffer, as temperature reading has 12 bits spanning two bytes:
+    buffer_rx[0] = 0;
+    buffer_rx[1] = 0;
+
+
+#ifdef DIAG_ON__TMP101AN
+    chprintf((BaseSequentialStream *) &SD2, "reading temperature sensor %02X . . .\r\n", sensor_addr);
+#endif
+
+    i2cAcquireBus(devp->config->i2cp);
+    i2cStart(devp->config->i2cp, devp->config->i2ccfg);
+
+    i2c_result = i2cMasterTransmitTimeout(devp->config->i2cp, sensor_addr, buffer_tx, 1, buffer_rx, 2, TIME_INFINITE);
+    i2c_flags = i2cGetErrors(devp->config->i2cp);
+    i2cReleaseBus(devp->config->i2cp);
+
+#ifdef DIAG_ON__TMP101AN
+    chprintf((BaseSequentialStream *) &SD2, "temperature reg bytes back hold %u and %u.\r\n\r\n",
+      buffer_rx[0], buffer_rx[1]);
+#endif
+
+    if ( i2c_result != 0 )
+        { chprintf((BaseSequentialStream *) &SD2, "i2c_result holds %ld.\r\n", i2c_result); }
+    if ( i2c_flags != 0 )
+        { chprintf((BaseSequentialStream *) &SD2, "i2c_flags holds %ld.\r\n\r\n", i2c_flags); }
+
+// Compiler error:  cannot write to instantiating code's memory, as on next line - TMH:
+//    devp->config->temperature_present_in_C = (( buffer_rx[0] << 4 ) | ( buffer_rx[0] >> 4 ));
+
+    if ( i2c_result == MSG_OK )
+        return true;
+    else
+        return true;
+
+}
+
+
+
+//----------------------------------------------------------------------
+// @brief   routine to read TMP101AN temperature sensor, and return two-
+//          two-byte reading
+//----------------------------------------------------------------------
+msg_t read_tmp101an_temperature_v2(TMP101Driver *devp, uint8_t* byte_array)
+{
+    msg_t i2c_result = MSG_OK;
+    i2cflags_t i2c_flags = 0;
+    uint8_t sensor_addr = devp->config->saddr;
+    uint8_t buffer_tx[2] = {0};
+
+// Values chosen per TMP101AN datasheet:
+    buffer_tx[0] = TMP101_REG_TEMPERATURE_READING;
+    buffer_tx[1] = 0;
+
+// Prepare for I2C transaction:
+    i2cAcquireBus(devp->config->i2cp);
```

Comment:
Same comments about conditional compilation based on TMP101_USE_I2C and TMP101_SHARED_BUS

Reply:
Fixed, I2C related conditionals now in place.